test/grpc_web_test.dart

```diff
@@ -116,14 +116,34 @@ void main() {
 
     await channel.terminate();
   });
+
+  final invalidResponseTests = {
+    'cors': GrpcError.unknown(
+        'HTTP request completed without a status (potential CORS issue)'),
+    'status-503': GrpcError.unavailable(
+        'HTTP connection completed with 503 instead of 200'),
+    'bad-content-type':
+        GrpcError.unknown('unsupported content-type (text/html)'),
+  };
+
+  for (var entry in invalidResponseTests.entries) {
+    test('invalid response: ${entry.key}', () async {
+      final channel = GrpcWebClientChannel.xhr(server.httpUri);
+      final service = EchoServiceClient(channel,
+          options: WebCallOptions(bypassCorsPreflight: true));
+      expect(() => service.echo(EchoRequest()..message = 'test:${entry.key}'),
```

Comment:
Maybe add a comment explaining that the server will treat `EchoRequest` with `test:` specially by returning errors instead of echoing anything.

Reply:
Done.